pkg/asset/cluster/tfvars.go

```diff
@@ -748,10 +748,19 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 			return err
 		}
 
-		// Get CISInstanceCRN from InstallConfig metadata
-		crn, err := installConfig.PowerVS.CISInstanceCRN(ctx)
-		if err != nil {
-			return err
+		var cisCRN, dnsCRN string
+		if installConfig.Config.Publish == types.InternalPublishingStrategy {
+			// Get DNSInstanceCRN from InstallConfig metadata
+			dnsCRN, err = installConfig.PowerVS.DNSInstanceCRN(ctx)
+			if err != nil {
+				return err
+			}
+		} else {
```

Comment:
i'm not sure, but is there a possiblity that this could be set to anything other than one of the two valid publishing types? or that it could be null? wondering if instead of juts if/else it should be a case statement with a default that's an error

Reply:
A third publishing strategy may be introduced one day, so will adjust